### PR TITLE
[full ci] Containers can connect to each other via mapped ports

### LINKED ISSUE #2295: Containers not able to connect to each other using exposed ports in a network
I am trying to get a simple Wordpress application running while making use of the networking feature. Basically I create a network, then create a mysql container, then create a wordpress container and point it to connect to the previously created mysql container. See Code section for details.

**VIC Version:** JFrog Bintray builds 4677, 4715

**Code**

```ts
# Create a network:
docker --tls network create -d bridge wpnet

# The create a mysql database. Note that the port 3306 is mapped to 3306 host port. This is done
# in order to use a single mysql container for 2 different (both failing) approaches to get the
# wordpress application running
docker --tls run -itd --name mysql -e MYSQL_ROOT_PASSWORD=pass@word01 --network=wpnet -p 3306:3306 mariadb:10.0.26

# Then create wordpress containers

# Approach 1 - use the docker host IP ($IP) and the mapped host port (3306)
docker --tls run -itd --name wordpress1 -e WORDPRESS_DB_PASSWORD=pass@word01 -e WORDPRESS_DB_HOST=$IP:3306 --network=wpnet -p 8081:80 wordpress:4.3.1

# Approach 2 - use the IP that the mysql container was assigned in the wpnet network
# (172.18.0.2, got this with docker network insect wpnet) and the port on which mysql is
# listening inside the container (3306 again)
docker --tls run -itd --name wordpress2 -e WORDPRESS_DB_PASSWORD=pass@word01 -e WORDPRESS_DB_HOST=172.18.0.2:3306 --network=wpnet -p 8082:80 wordpress:4.3.1
```

**Expected behavior:**
Wordpress container successfully connecting to mysql container in both cases and showing the installation wizard in web browser when accessed on $IP:8081 and $IP:8082 respectively.

**Actual behavior:**
- Approach 1 - Wordpress fails to connect to mysql container and thus refuses HTTP connections
```ts
$ docker --tls logs wordpress1
WordPress not found in /var/www/html - copying now...
WARNING: /var/www/html is not empty - press Ctrl+C now if this is an error!
+ ls -A
.wh..wh..opq  index.html
+ sleep 10
Complete! WordPress has been successfully copied to /var/www/html

Warning: mysqli::mysqli(): (HY000/2002): Connection refused in - on line 10

MySQL Connection Error: (2002) Connection refused

Warning: mysqli::mysqli(): (HY000/2002): Connection refused in - on line 10

MySQL Connection Error: (2002) Connection refused

Warning: mysqli::mysqli(): (HY000/2002): Connection refused in - on line 10

MySQL Connection Error: (2002) Connection refused
```
- Approach 2 - Apache server in wordpress container returns a configuration error, thus wordpress is refusing HTTP connections once again:
```
$ docker --tls logs wordpress2
WordPress not found in /var/www/html - copying now...
WARNING: /var/www/html is not empty - press Ctrl+C now if this is an error!
+ ls -A
.wh..wh..opq  index.html
+ sleep 10
Complete! WordPress has been successfully copied to /var/www/html
[Sat Sep 10 11:00:21.626517 2016] [core:warn] [pid 200:tid 139777220556672] AH00111: Config variable ${APACHE_LOG_DIR} is not defined
[Sat Sep 10 11:00:21.651821 2016] [core:warn] [pid 200:tid 139777220556672] AH00111: Config variable ${APACHE_LOG_DIR} is not defined
[Sat Sep 10 11:00:21.651868 2016] [core:warn] [pid 200:tid 139777220556672] AH00111: Config variable ${APACHE_LOG_DIR} is not defined
AH00534: apache2: Configuration error: More than one MPM loaded.
```

**Note:**
Initially, I tried to set the application "the right way", i.e., I was passing `-e WORDPRESS_DB_HOST=mysql:3306` when starting the wordpress container. However, it looks like the names of the containers are not resolved currently (see #2294), so I fell back to these approaches.

### PR DESCRIPTION
**Pull request by andrewtchin**
Adds rule to allow traffic from one container to another through mapped port on the VCH external IP
```
root@chin-vic-10 [ ~ ]# iptables -L -n -v -t nat
Chain PREROUTING (policy ACCEPT 2 packets, 230 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    8   468 VIC        all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT 1 packets, 60 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain OUTPUT (policy ACCEPT 1 packets, 76 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 VIC        all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
   91  5274 MASQUERADE  all  --  *      !bridge  0.0.0.0/0            0.0.0.0/0           
    0     0 MASQUERADE  tcp  --  *      *       0.0.0.0/0            172.16.0.2           tcp dpt:80

Chain VIC (2 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DNAT       tcp  --  client *       0.0.0.0/0            0.0.0.0/0            tcp dpt:8083 to:172.16.0.2:80
    0     0 DNAT       tcp  --  bridge *       0.0.0.0/0            0.0.0.0/0            tcp dpt:8083 ADDRTYPE match dst-type LOCAL to:172.16.0.2:80
```

Fixes #2295

### CODE REVIEW

File: lib/apiservers/engine/backends/portmap/portmap.go [hunk 1]
```diff
@@ -181,6 +181,21 @@ func (p *portMapper) forward(action iptables.Action, ip net.IP, port int, proto,
...
+		// allow traffic from container to container via vch external interface
+		args = []string{"VIC", "-t", string(iptables.Nat),
+			"-i", destIface,
+			"-p", proto,
+			"--dport", strconv.Itoa(port),
+			"-j", "DNAT",
+			"--to-destination", net.JoinHostPort(destAddr, strconv.Itoa(destPort)),
+			"-m", "addrtype",
+			"--dst-type", "LOCAL"}
+		if err := iptablesRunAndCheck(action, args); err != nil {
```
Comment (vburenin) [inline lib/apiservers/engine/backends/portmap/portmap.go]: Is there any rollback needed if attempt to run iptables fails?

**Comment by vburenin** [comment]
LGTM

**Comment by sgairo** [comment]
LGTM

File: tests/test-cases/Group1-Docker-Commands/1-25-Docker-Port-Map.robot [hunk 2]
```diff
@@ -120,3 +120,43 @@ Remap mapped ports after OOB Stop
...
+    Should Not Contain  ${output}  Error
+
+    Sleep  10
+
+    ${rc}  ${ip}=  Run And Return Rc And Output  docker ${params} network inspect bridge | jq '.[0].Containers."${containerID}".IPv4Address'
+    ${ip}=  Split String  ${ip}  /
+    ${nginx-ip}=  Set Variable  @{ip}[0]
+    ${nginx-ip}=  Strip String  ${nginx-ip}  characters="
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} run --name anjunabeats busybox /bin/ash -c "wget -O index.html ${ext-ip}:8085; md5sum index.html"
```
Comment (anchal-agrawal) [inline tests/test-cases/Group1-Docker-Commands/1-25-Docker-Port-Map.robot]: +1 for the container name.

**Comment by hmahmood** [comment]
lgtm

File: tests/test-cases/Group1-Docker-Commands/1-25-Docker-Port-Map.robot (hunk 2 above)
Comment (andrewtchin) [inline tests/test-cases/Group1-Docker-Commands/1-25-Docker-Port-Map.robot]: lol :-)

**Comment by andrewtchin** [comment]
♻️

**Pull request merged**